### ISSUE TITLE
Problem sending files with correct MIME type without file extension

### ISSUE THREAD

**Issue by jd-boyd** [civetweb/civetweb#271]
I have files with machine generated names and no file extension that I want to serve from a customer handler with `mg_send_file`.  The problem is that without a file extension, `mg_send_file` assumes the wrong MIME type.

I would like to see a new function added like `mg_send_file`, but that takes a MIME type as an argument, perhaps:
```c
void
mg_send_file_with_mime_type(struct mg_connection *conn, 
                            const char *path, 
                            const char *mime_type)
```

That in turn would probably require a change to `handle_static_file_request`.  Since that it static (and thus changes to it won't break outside users), I'd suggest it is reasonable to change its signature by adding a MIME type argument to it instead of making a new function.  Thus:
```c
static void
handle_static_file_request(struct mg_connection *conn,
                           const char *path,
                           struct file *filep,
                           const char *mime_type)
```
If `mime_type` is null, `handle_static_file_request` would behave as it currently does, but if it is supplied, then it would use the supplied MIME type instead of calling `get_mime_type`.

I'd be willing to implement this as described, but I wanted to submit the idea for feedback before expending the effort.

**Comment by bel2125** [comment]
The idea sounds reasonable.
I would not use `handle_static_file_request` directly, since it bypasses the 404 from `mg_send_file`.

I would:
keep `handle_static_file_request` static, 
add a `const char * mime_type` parameter, 
threat `mime_type == NULL` vs. `!= NULL` like you said,
rename `mg_send_file` in civetweb.c to `mg_send_mime_file`,
add the `const char * mime_type` parameter as well,
add a new wrapper function `mg_send_file` just calling `mg_send_mime_file` with the `mime_type` parameter `NULL`
Add the `mime_type` parameter `NULL` to any other `handle_static_file_request` call
Add  `mg_send_mime_file` to civetweb.h with a proper interface description.

What do you think?

**Comment by jd-boyd** [comment]
Sounds good.

**Comment by bel2125** [comment]
Solved with #275.
